docs/src/concepts.md

```diff
@@ -0,0 +1,123 @@
+
+## Formatting [new]
+
+Formatting is the art of computing a minimal set of white space text replacements. Formatting may be applied manually through a command invocation, or automatically when saving the file. Sublime Text has no concept of Formatting.
+
+## Signature Help [new]
+
+Signature Help is similar to Auto Complete, but focuses on presenting the different overloads of a function invocation. This package provides Signature Help by means of showing a popup. If multiple overloads exist, you can browse through them just like Auto Complete. Sublime Text has no concept of Signature Help.
+
+## Rename [new]
+
+When you want to rename an identifier in Sublime Text, you probably use <kbd>ctrl</kbd><kbd>D</kbd> to select a few next occurences and do the rename with multiple cursors.
+
+Because a language server (usually) has an abstract syntax tree view of the file, it may be able to rename an identifier semantically. This package exposes that functionality through the hover popup, the context menu, and the top menu bar.
+
+Some language servers provide _global_ rename functionality as well. This package will present a modal dialog to ask you to confirm to apply the changes if they span more than one file.
+
+## Code Actions [new]
+
+Code Actions are an umbrella term for "Quick Fixes" and "Refactorings". They are actions that change the file (or more than one file) to resolve a diagnostic or apply a standard refactor technique. For instance, extracting a block of code into a separate method is usually called "Extract Method" and is a "Refactoring". Whereas "add a missing semicolon" would resolve a diagnostic that warns about a missing semicolon.
+
+Formatting is different from Code Actions, because Formatting is supposed to _not_ mutate the abstract syntax tree of the file, only move around white space. Any Code Action will mutate the abstract syntax tree.
+
+This package presents Code Actions as a bluish clickable annotation positioned to the right of the viewport. Alternatively, they can be presented as a light bulb in the Gutter Area.
+
+Sublime Text has no concept of Code Actions.
+
+## Code Lenses [new]
+
+Code Lenses are "actionable contextual information interspersed" in your source code.
+
+- Actionable: You can click on the link and something happens.
+- Contextual: The links are close to the code they are representing.
+- Interspersed: The links are located throughout your source code.
+
+This package presents Code Lenses as a greenish clickable annotation positioned to the right of the viewport. Alternatively, they can be presented as phantoms.
+
+Sublime Text has no concept of Code Lenses.
+
+## Server Commands
+
+In Sublime Text you can bind any runnable command to a key or add it to various UI elements. Commands in Sublime Text are normally written in the Python plugin host. A language server may provide a runnable command as well. These kinds of commands are wrapped in an `lsp_execute` Sublime command that you can bind to a key.
```

Comment:
> Commands in Sublime Text are normally written in the Python plugin host.

Doesn't sound right. They are written in the editor.
Maybe: "Commands in Sublime Text are normally supplied by plugins or packages written in Python.